src/Compilers/CSharp/Portable/Compilation/BuiltInOperators.cs

```diff
@@ -696,15 +696,16 @@ private TypeSymbol LiftedType(BinaryOperatorKind kind)
             return null;
         }
 
-        internal static bool IsValidObjectEquality(Conversions Conversions, TypeSymbol leftType, bool leftIsNull, TypeSymbol rightType, bool rightIsNull, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        internal static bool IsValidObjectEquality(Conversions Conversions, TypeSymbol leftType, bool leftIsNull, bool leftIsDefault, TypeSymbol rightType, bool rightIsNull, bool rightIsDefault, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
         {
             // SPEC: The predefined reference type equality operators require one of the following:
 
-            // SPEC: (1) Both operands are a value of a type known to be a reference-type or the literal null. 
-            // SPEC:     Furthermore, an explicit reference conversion exists from the type of either 
+            // SPEC: (1) Both operands are a value of a type known to be a reference-type or the literal null.
+            // SPEC:     Furthermore, an explicit reference conversion exists from the type of either
             // SPEC:     operand to the type of the other operand. Or:
             // SPEC: (2) One operand is a value of type T where T is a type-parameter and the other operand is 
             // SPEC:     the literal null. Furthermore T does not have the value type constraint.
+            // SPEC: (3) One operand is the literal default and the other operand is a reference-type.
```

Comment:
Did the spec actually change?

Reply:
It hasn't yet, but I want to record the proposed spec change. I don't know a better way to keep the code and comment in sync...